deploy-agent/deployd/common/stats.py

```diff
@@ -28,69 +49,287 @@ def __exit__(self, stats, sample_rate=1.0, tags=None):
+            def timing(*args, **kwargs):
+                client = MetricClient()
+                return client.send_context_timer(*args, **kwargs)
 
-def create_stats_timer(stats, sample_rate=1.0, tags=None):
-    if IS_PINTEREST:
-        try:
-            from pinstatsd.statsd import statsd_context_timer
-            tags = _add_default_tags(tags)
-            return statsd_context_timer(entry_name=stats, sample_rate=sample_rate, tags=tags)
-        except Exception as e:
-            error_msg = str(e)
-            log.error('unable to send metric: {}'.format(error_msg))
+        timer = statsd_context_timer(entry_name=name, sample_rate=sample_rate, tags=tags)
+        timer._statsd_context_timer__stat_client = TimingOnlyStatClient()
+        return timer
     else:
-        return DefaultStatsdTimer()
+        return
 
 
-def create_sc_timing(stat, value, sample_rate=1.0, tags=None):
+def create_sc_timing(name, value, sample_rate=1.0, tags=None):
     if IS_PINTEREST:
-        try:
-            from pinstatsd.statsd import sc
-            tags = _add_default_tags(tags)
-            return sc.timing(stat, value, sample_rate=sample_rate, tags=tags)
-        except Exception as e:
-            error_msg = str(e)
-            log.error('unable to send metric: {}'.format(error_msg))
+        mtype = 'timing'
+        client = MetricClient()
+        client.send(mtype=mtype,
+                    name=name,
+                    value=value,
+                    sample_rate=sample_rate,
+                    tags=tags)
     else:
         return
 
 
-def create_sc_increment(stats, sample_rate=1.0, tags=None):
+def create_sc_increment(name, sample_rate=1.0, tags=None):
     if IS_PINTEREST:
-        try:
-            from pinstatsd.statsd import sc
-            tags = _add_default_tags(tags)
-            sc.increment(stats, sample_rate, tags)
-        except Exception as e:
-            error_msg = str(e)
-            log.error('unable to send metric: {}'.format(error_msg))
+        mtype = 'increment'
+        client = MetricClient()
+        client.send(mtype=mtype,
+                    name=name,
+                    sample_rate=sample_rate,
+                    tags=tags)
     else:
         return
 
 
-def create_sc_gauge(stat, value, sample_rate=1.0, tags=None):
+def create_sc_gauge(name, value, sample_rate=1.0, tags=None):
     if IS_PINTEREST:
-        try:
-            from pinstatsd.statsd import sc
-            tags = _add_default_tags(tags)
-            sc.gauge(stat, value, sample_rate=sample_rate, tags=tags)
-        except Exception as e:
-            error_msg = str(e)
-            log.error('unable to send metric: {}'.format(error_msg))
+        mtype = 'gauge'
+        client = MetricClient()
+        client.send(mtype=mtype,
+                    name=name,
+                    value=value,
+                    sample_rate=sample_rate,
+                    tags=tags)
     else:
         return
 
 
+class Cache:
```

Comment:
rename is to MetricsCache.

Reply:
Great suggestion.  Done!  That makes it more readable.